code/game/atoms_movable.dm

```diff
@@ -719,3 +719,46 @@
 
 /atom/movable/proc/can_apply_inertia()
 	return (!src.anchored && !(src.pulledby && src.pulledby.Adjacent(src)))
+
+/atom/movable/proc/send_to_future(var/duration)
+	var/init_invisibility = invisibility
+	var/init_invuln = flags & INVULNERABLE
+	var/init_density = density
+	var/init_anchored = anchored
+	var/init_timeless = flags & TIMELESS
+	var/init_blinded
+	var/mob/M
+	if(ismob(src))
```

Comment:
In testing, it seems that the child proc does *not* wait for the parent proc to finish before continuing past the `..()`, if the parent proc has `set waitfor = 0`.
So either I need to have the src typecheck, or copy-paste the whole proc into the child instead of calling `..()`, or split the code before and after the `spawn()` into two different procs and override both of them.